src/index.ts

```diff
@@ -8,14 +8,27 @@ export interface IConfig {
     clientKey: string;
     appName: string;
     environment?: string;
+    userId?: string;
```

Comment:
it feels a bit cumbersome to have all context fields also defined as part of the config. The initial idea was that we have a few "static context fields" (`appName` and `envrionment`) which will never change, whereas the other context fields is more dynamic and can change in the lifetime of an application (e.g.: a user may login without a reload, or may be part of a new group after some actions). 

Maybe we could have a IStaticContext interface interface and have IContext inherit that and have a property on the config interface for the static context fields? What do you think?

Reply:
Ah absolutely I agree. It does feel cumbersome having all the context fields there. Let me clean it up a bit with these suggestions.

Reply:
@ivarconr see e16704a. This would be a breaking change though and require the consumer to define the Unleash Context and pass it in. Might not be what we want, but I wanted to ask your opinion. I think the types are a lot cleaner, but this we might want to keep the same UnleaseClient api.

Reply:
8cb9aa9 This is now a non-breaking change, while maintaining the advantages of the previous commits. I still think the previous breaking change is more in line with Unleash Context, but in the interest of speed and no breaking changes, I think this is best

Reply:
I agree with your latest change. Lets try not to do a breaking change for now. We can always do that later.